### PR TITLE
Document confusing module test behavior

### PR DESCRIPTION
**Pull request by JacobEvelyn**
**Before merging:**

- [x] If this change merits an update to `CHANGELOG.md`, add an entry following Keep a Changelog [guidelines](https://keepachangelog.com/en/1.0.0/) with [semantic versioning](https://semver.org/)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,6 +140,25 @@ code examples in our YARD documentation. To run `doctest` locally:
 bundle exec yard doctest
 ```
 
+### A Note on Testing
```
Comment (JacobEvelyn) [inline README.md:143]: I couldn't figure out a good place in YARD to put this but let me know if there is!

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-(nothing yet!)
+- Documentation of confusing module test behavior
```
[the remark this answers is not in the document]
Comment (ms-ati) [reply CHANGELOG.md:9]: Seems fine to me

**Review by ms-ati — APPROVED** [review]
LGTM!

**Review by ms-ati — APPROVED** [review]
LGTM!

File: README.md [hunk 3]
```diff
@@ -141,6 +140,25 @@ code examples in our YARD documentation. To run `doctest` locally:
...
+across tests, which can result in confusing test failures when this differs from
+how you use the code in production.
+
+For example, Rails view helpers are modules that are commonly tested with a
+[shared `view` instance](https://github.com/rails/rails/blob/291a3d2ef29a3842d1156ada7526f4ee60dd2b59/actionview/lib/action_view/test_case.rb#L203-L214). Rails initializes a new view instance for each web request so any view helper
+methods would only be memoized for the duration of that web request, but in
+tests (such as when using
+[`rspec-rails`'s `helper`](https://github.com/rspec/rspec-rails/blob/main/lib/rspec/rails/example/helper_example_group.rb#L22-L27)),
+the memoization may persist across tests. In this case, simply run something
+like this between your tests:
```
Comment (jemmaissroff) [inline README.md:156]: Can we be more explicit and say in a `before_each` or `after_each` block?

**Review by jemmaissroff — CHANGES_REQUESTED** [review]

**Review by ms-ati — APPROVED** [review]
LGTM

**Review by jemmaissroff — APPROVED** [review]

**Pull request merged**